test/runner/browserstack/queue.js

```diff
@@ -0,0 +1,81 @@
+import chalk from "chalk";
+import { getBrowserString } from "../lib/getBrowserString.js";
+import { checkLastTouches, createBrowserWorker, setBrowserWorkerUrl } from "./browsers.js";
+
+const TEST_POLL_TIMEOUT = 1000;
+
+const queue = [];
+
+export function getNextBrowserTest( reportId ) {
+	const index = queue.findIndex( ( test ) => test.id === reportId );
+	if ( index === -1 ) {
+		return;
+	}
+	const previousTest = queue[ index ];
+	queue.splice( index, 1 );
+	for ( const test of queue.slice( index ) ) {
+		if ( test.fullBrowser === previousTest.fullBrowser ) {
+			setBrowserWorkerUrl( test.browser, test.url );
+			test.running = true;
+			return { url: test.url };
+		}
+	}
+}
```

Comment:
It'd be good to have some comments for this function. Especially about this browser reuse, why we look at the tests after the one with `reportId`, etc.